deployments/helm/nvidia-dra-driver-gpu/templates/_helpers.tpl

```diff
@@ -66,14 +66,19 @@ app.kubernetes.io/instance: {{ .Release.Name }}
 {{- end }}
 
 {{/*
-Selector labels
+Selector labels: meant to select just the pods of the corresponding
+deployment/daemonset/... That is, this is different per-component.
 */}}
 {{- define "nvidia-dra-driver-gpu.selectorLabels" -}}
-{{- if .Values.selectorLabelsOverride -}}
-{{ toYaml .Values.selectorLabelsOverride }}
-{{- else -}}
-{{ include "nvidia-dra-driver-gpu.templateLabels" . }}
-{{- end }}
+  {{- if and (hasKey . "componentName") (hasKey . "context") -}}
+    {{- if .context.Values.selectorLabelsOverride -}}
+    {{ toYaml .context.Values.selectorLabelsOverride }}
+    {{- else -}}
+    nvidia-dra-driver-gpu-component: {{ .componentName }}
+    {{- end }}
+  {{- else -}}
+  fail "selectorLabels: both arguments are required: context, componentName"
```

Comment:
I adopted this approach from some blog post: named and required function arguments, with a more or less good error message upon not providing arguments.

Note that the outer chart context _in the function_ is now not `.` anymore, but `.context`. 

Don't we love learning DSLs...and their funkiness :)

refs:
https://blog.palark.com/advanced-helm-templating/
https://jiminbyun.medium.com/understanding-the-chart-context-in-helm-templates-0524b520b120

Reply:
Just to clarify: The `fail` here just checks that we're "invoking" this helper correctly? Meaning we need to set both the context and the component name where we include it in some other helm template.

Reply:
Yes! It's like an assert statement.